[Inventory] Create new Inventory docs (#4341)
elastic/observability-docs: docs/en/serverless/inventory.mdx

---
slug: /serverless/observability/inventory
title: Inventory
description: Learn about the new Inventory experience that enables you to monitor all your entities from one single place.
tags: [ 'serverless', 'observability', 'inventory' ]
---

import Roles from './partials/roles.mdx'

<p><DocBadge template="technical preview" /></p>

Inventory provides a single place where you can observe the status of your entire ecosystem of hosts, containers and services at a glance, even just from logs. From there, you can monitor and understand the health of your entities, check what needs attention and start your investigations. 

<DocCallOut title="Note">
   The new Inventory requires the Elastic Entity Model (EEM). To learn more, refer to <DocLink slug="/serverless/observability/elastic-entity-model" />.
</DocCallOut>

> <DocImage url="../images/inventory-catalog.png" alt="Inventory catalog"/>

Reviewer: ```suggestion
<DocImage url="./images/inventory-catalog.png" alt="Inventory catalog"/>
```

Inventory allows you to:

- Filter for your entities to provide a high-level view of what you have leveraging your own tags and labels
- Drill down into any host, container or services to help you understand performance
- Debug resource bottlenecks with your service caused by their containers and the hosts they run on.
- Easily discover all entities related to the host, container or service you are viewing by leveraging your tags/labels

## Explore your entities

1. In your ((observability)) project, go to **Inventory** to open the entities catalog.

    If you have **Admin** privileges, the EEM will be turned on automatically. Otherwise, you need to ask someone with **Admin** privileges to turn on this feature under **Stack Management**.

1. In the search bar, search for your entities by name or type, for example `entity.type:service`.

For each entity, you can either view your data in Discover, or click the entity name and get a detailed view. For example, for an entity of type `service`, you get the following details:

- Overview
- Transactions
- Dependencies
- Errors
- Metrics
- Infrastructure
- Service Map
- Logs
- Alerts
- Dashboards

> <DocImage url="../images/entity-detailed-view.png" alt="Entity detailed view"/>

Reviewer: ```suggestion
<DocImage url="./images/entity-detailed-view.png" alt="Entity detailed view"/>
```

---
slug: /serverless/observability/inventory
title: Inventory
description: Learn about the new Inventory experience that enables you to monitor all your entities from one single place.
tags: [ 'serverless', 'observability', 'inventory' ]
---

import Roles from './partials/roles.mdx'

<p><DocBadge template="technical preview" /></p>

> Inventory provides a single place where you can observe the status of your entire ecosystem of hosts, containers and services at a glance, even just from logs. From there, you can monitor and understand the health of your entities, check what needs attention and start your investigations. 

Author: "even just from logs" is the added value :-) I would leave it as is for now.

<DocCallOut title="Note">
   The new Inventory requires the Elastic Entity Model (EEM). To learn more, refer to <DocLink slug="/serverless/observability/elastic-entity-model" />.
</DocCallOut>

<DocImage url="./images/inventory-catalog.png" alt="Inventory catalog"/>

> Inventory is currently available for hosts, containers and services, but it will scale to support all of your entities.

Reviewer 2: ```suggestion
Inventory is currently available for hosts, containers, and services, but it will scale to support all of your entities.
```

> The Elastic Entity Model currently supports the inventory experience (as identified by `host.name`, `service.name`, and `container.id`) located in data identified by the following index patterns:

Reviewer 2: EEM is called out previously, so you can probably use the abbreviation here.

```suggestion
The EEM currently supports the inventory experience (as identified by `host.name`, `service.name`, and `container.id`) located in data identified by the following index patterns:
```

**Hosts**

Where `host.name` is set in `metrics-*`, `logs-*`, `filebeat-*`, and `metricbeat-*`

**Services**

Where `service.name` is set in `filebeat*`, `logs-*`, `metrics-apm.service_transaction.1m*`, and `metrics-apm.service_summary.1m*`
  
**Containers**

Where `container.id` is set in `metrics-*`, `logs-*`, `filebeat-*`, and `metricbeat-*`

Inventory allows you to:

- Filter for your entities to provide a high-level view of what you have leveraging your own tags and labels

> - Drill down into any host, container or services to help you understand performance

Reviewer 2: ```suggestion
- Drill down into any host, container, or service to help you understand performance
```

- Debug resource bottlenecks with your service caused by their containers and the hosts they run on.

> - Easily discover all entities related to the host, container or service you are viewing by leveraging your tags/labels

Reviewer 2: ```suggestion
- Easily discover all entities related to the host, container or service you are viewing by leveraging your tags and labels
```

## Explore your entities

1. In your ((observability)) project, go to **Inventory** to view all of your entities.

>     When you open the Inventory for the first time, you need to enable the Elastic Entity Model to start using it. Once enabled, the Inventory will be accessible to anyone with the appropriate privileges.

Reviewer 2: Is this something that will be obvious to users? Will it be like an empty state page that tells them they need to enable EEM? If so, could we say:
```suggestion
    When you open the Inventory for the first time, you'll be asked to enable the EEM. Once enabled, the Inventory will be accessible to anyone with the appropriate privileges.
```

    <DocCallOut title="Note">
    The Inventory feature can be completely disabled using the `observability:entityCentricExperience` flag in **Stack Management**.    
    </DocCallOut>
    

1. In the search bar, search for your entities by name or type, for example `entity.type:service`.

For each entity, you can click the entity name and get a detailed view. For example, for an entity of type `service`, you get the following details:

- Overview
- Transactions
- Dependencies
- Errors
- Metrics
- Infrastructure
- Service Map
- Logs
- Alerts
- Dashboards

<DocImage url="./images/entity-detailed-view.png" alt="Entity detailed view"/>

If you open an entity of type `host` or `container` that does not have infrastructure data, some of the visualizations will be blank and some features on the page will not be fully populated.

> ## Add entities to the inventory

Reviewer 2: ```suggestion
## Add entities to the Inventory
```

> Entities are added to the inventory through one of the following approaches: **Add data** or **Associate existing service logs**.

Reviewer 2: ```suggestion
Entities are added to the Inventory through one of the following approaches: **Add data** or **Associate existing service logs**.
```

### Add data

> To add entities, select **Add data** from the left-hand navigation and choose one the following onboarding journeys:

Reviewer 2: ```suggestion
To add entities, select **Add data** from the left-hand navigation and choose one of the following onboarding journeys:
```

<DocDefTerm>- Auto-detect logs and metrics</DocDefTerm>
<DocDefDescription>
 Detects hosts (with metrics and logs)    
</DocDefDescription>

<DocDefTerm>- Kubernetes</DocDefTerm>
<DocDefDescription>

>  Detects hosts, containers and services    

Reviewer 2: ```suggestion
 Detects hosts, containers, and services    
```
